roles/confluent.kafka_broker/templates/server.properties.j2

```diff
@@ -9,43 +9,55 @@ broker.id={{ broker_id | default(groups.kafka_broker.index(inventory_hostname) +
+listener.security.protocol.map={{kafka_broker_internal_listener_name}}:{{kafka_broker_internal_listener_security_protocol}},{{kafka_broker_external_listener_name}}:{{kafka_broker_external_listener_security_protocol}}
+
+## Inter Broker Listener Configuration
+inter.broker.listener.name={{kafka_broker_internal_listener_name}}
+{% if kafka_broker_internal_listener_sasl_protocol_normalized != 'none' %}
+sasl.mechanism.inter.broker.protocol={{kafka_broker_internal_listener_sasl_protocol_normalized}}
+{% endif %}
+{% if kafka_broker_sasl_enabled_mechanisms|length > 0 %}
+sasl.enabled.mechanisms={% for mechanism in kafka_broker_sasl_enabled_mechanisms %}{% if loop.index > 1%},{% endif %}{{ mechanism|upper }}{% endfor %}
+
+{% endif %}
+{% if 'GSSAPI' in kafka_broker_sasl_enabled_mechanisms %}
 sasl.kerberos.service.name={{kerberos_kafka_broker_primary}}
 {% endif %}
+{% with %}
+{% set listener_name = kafka_broker_internal_listener_name %}
+{% set listener_ssl_enabled = kafka_broker_internal_listener_ssl_enabled %}
+{% set listener_ssl_mutual_auth_enabled = kafka_broker_internal_listener_ssl_mutual_auth_enabled %}
+{% set listener_sasl_protocol = kafka_broker_internal_listener_sasl_protocol_normalized %}
+{% include 'listener.j2' %}
+{% endwith %}
 
-confluent.metrics.reporter.security.protocol={{kafka_broker_security_protocol}}
-{% if kafka_broker_ssl_enabled|bool %}
+## External Client Listener Configuration
+{% with %}
+{% set listener_name = kafka_broker_external_listener_name %}
+{% set listener_ssl_enabled = kafka_broker_external_listener_ssl_enabled %}
+{% set listener_ssl_mutual_auth_enabled = kafka_broker_external_listener_ssl_mutual_auth_enabled %}
+{% set listener_sasl_protocol = kafka_broker_external_listener_sasl_protocol_normalized %}
+{% include 'listener.j2' %}
+{% endwith %}
+
+## Metrics Reporter Configuration
+confluent.metrics.reporter.security.protocol={{kafka_broker_external_listener_security_protocol}}
+{% if kafka_broker_external_listener_ssl_enabled|bool %}
 confluent.metrics.reporter.ssl.truststore.location={{truststore_path}}
 confluent.metrics.reporter.ssl.truststore.password={{truststore_storepass}}
+{% if kafka_broker_external_listener_ssl_mutual_auth_enabled|bool %}
 confluent.metrics.reporter.ssl.keystore.location={{keystore_path}}
 confluent.metrics.reporter.ssl.keystore.password={{keystore_storepass}}
 confluent.metrics.reporter.ssl.key.password={{keystore_keypass}}
 {% endif %}
-{% if sasl_protocol == 'plain' %}
+{% endif %}
+{% if kafka_broker_external_listener_sasl_protocol_normalized == 'PLAIN' %}
 confluent.metrics.reporter.sasl.mechanism=PLAIN
-confluent.metrics.reporter.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="client" password="client-secret";
+confluent.metrics.reporter.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
+   username="{{sasl_plain_users.admin.principal}}" password="{{sasl_plain_users.admin.password}}";
```

Comment:
@JumaX What user do you think the metrics reporter on the broker be calling in as?
I have it as admin rn for plain and scram

Reply:
@domenicbove good question.  Typically we see the customer use the same user as what they are using to authorize the Brokers.